src/QirRuntime/samples/StandaloneInputReference/qir-driver.cpp

```diff
@@ -0,0 +1,269 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <cstring> // for memcpy
+#include <fstream>
+#include <iostream>
+#include <map>
+#include <vector>
+
+#include "CoreTypes.hpp"
+#include "QirContext.hpp"
+#include "QirTypes.hpp"
+#include "QuantumApi_I.hpp"
+#include "SimFactory.hpp"
+
+#include "quantum__qis_internal.hpp"
+#include "quantum__rt.hpp"
+
+#include "CLI11.hpp"
+
+using namespace Microsoft::Quantum;
+using namespace std;
+
+// This is the function corresponding to the QIR entry-point.
+extern "C" int64_t Quantum__StandaloneSupportedInputs__ExerciseInputs__body( // NOLINT
+    int64_t intValue,
+    double doubleValue,
+    Result resultValue,
+    QirString* stringValue,
+    int64_t rangeStart,
+    int64_t rangeStep,
+    int64_t rangeEnd);
+
+const char FalseAsChar = 0x0;
+const char TrueAsChar = 0x1;
+map<string, bool> BoolAsCharMap{
+    {"0", FalseAsChar},
+    {"false", FalseAsChar},
+    {"1", TrueAsChar},
+    {"true", TrueAsChar}};
+
+map<string, PauliId> PauliMap{
+    {"PauliI", PauliId::PauliId_I},
+    {"PauliX", PauliId::PauliId_X},
+    {"PauliY", PauliId::PauliId_Y},
+    {"PauliZ", PauliId::PauliId_Z}};
+
+map<string, char> ResultAsCharMap{
+    {"0", FalseAsChar},
+    {"Zero", FalseAsChar},
+    {"false", FalseAsChar},
+    {"1", TrueAsChar},
+    {"One", TrueAsChar},
+    {"true", TrueAsChar}};
+
+template<typename T>
+QirArray* CreateQirArray(T* dataBuffer, int64_t itemCount)
+{
+    int32_t typeSize = sizeof(T); // NOLINT
+    QirArray* qirArray = quantum__rt__array_create_1d(typeSize, itemCount);
+    memcpy(qirArray->buffer, dataBuffer, typeSize * itemCount);
+    return qirArray;
+}
+
+template<typename D, typename S>
+void TranslateVectorToBuffer(D** destinationBuffer, vector<S>sourceVector, function<D(S)> translationFunction)
+{
+    *destinationBuffer = new D[sourceVector.size()];
+    for (int index = 0; index < sourceVector.size(); index++)
+    {
+        (*destinationBuffer)[index] = translationFunction(sourceVector[index]);
+    }
+}
```

Comment:
(If you have time and desire)

To make a step to more memory-safe code, you may want to consider returning a `unique_ptr<D>` instead of updating the `*destinationBuffer`.
Otherwise currently in the fragments like this
```c++
    bool* boolArray = nullptr;
    TranslateVectorToBuffer<bool, char>(&boolArray . . .
```
the `boolArray` is not deallocated. 
If it was happening not in `main()` but in some other function then we would be leaking that mem.

The change could be something like this:
> template<typename D, typename S>
**unique_ptr\<D>** TranslateVectorToBuffer(~~D** destinationBuffer,~~ vector\<S>sourceVector,
{
    **unique_ptr\<D> retVal {** new D[sourceVector.size()] };

Use:
> **unique_ptr\<bool>**  boolArray = ~~nullptr;~~
TranslateVectorToBuffer<bool, char>( ~~\&boolArray,~~ boolAsCharVector, TranslateCharToBool);
QirArray* qirboolArray = CreateQirArray(boolArray **.get()**, boolAsCharVector.size());